dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/fileresource/FileResourceCleanUpJob.java

```diff
@@ -54,14 +55,29 @@
 public class FileResourceCleanUpJob
     extends AbstractJob
 {
-    @Autowired
-    private FileResourceService fileResourceService;
+    private final FileResourceService fileResourceService;
 
-    @Autowired
-    private SystemSettingManager systemSettingManager;
+    private final SystemSettingManager systemSettingManager;
 
-    @Autowired
-    private FileResourceContentStore fileResourceContentStore;
+    private final FileResourceContentStore fileResourceContentStore;
+
+    // -------------------------------------------------------------------------
+    // Consructor
+    // -------------------------------------------------------------------------
+
+    public FileResourceCleanUpJob(
+        FileResourceService fileResourceService,
+        SystemSettingManager systemSettingManager,
+        FileResourceContentStore fileResourceContentStore )
+    {
+        checkNotNull( fileResourceService );
+        checkNotNull( systemSettingManager );
+        checkNotNull( fileResourceContentStore );
+
+        this.fileResourceService = fileResourceService;
+        this.systemSettingManager = systemSettingManager;
+        this.fileResourceContentStore = fileResourceContentStore;
+    }
```

Comment:
FYI: Such initialisation of beans can be done using `@AllArgsConstructor` to spare us having to read and write the boilerplate code.

Reply:
@jbee Thanks for the tip. Done!